store.go

```diff
@@ -1887,11 +1887,14 @@ func (s *store) layersByMappedDigest(m func(ROLayerStore, digest.Digest) ([]Laye
 			store.Load()
 		}
 		storeLayers, err := m(store, d)
-		if err != nil {
+		if err != nil && errors.Cause(err) != ErrLayerUnknown {
 			return nil, err
 		}
 		layers = append(layers, storeLayers...)
```

Comment:
Should you still be appending if you get ErrLayernUnknown?

Reply:
Hmm, probably not.  Revised.